### ISSUE TITLE
lsp: `deno/cache` requests respond the same for success or failure

### ISSUE THREAD

**Issue by jaydenseric** [denoland/deno#11796]
A Deno LSP [`deno/cache`](https://github.com/denoland/deno/tree/main/cli/lsp#requests) request always responds the same, whether the operation was a success or failure.

For example, if you have a file called `foo.js` containing an uncached import, and you disconnect your computer from the internet, then attempt a `deno/cache` request…

LSP request:

```json
{
  "jsonrpc": "2.0",
  "id": 70,
  "method": "deno\\/cache",
  "params": {
    "referrer": {
      "uri": "file:\\/\\/[redacted]\\/foo.js"
    },
    "uris": []
  }
}
```

LSP response:

```json
{
  "jsonrpc": "2.0",
  "id": 70,
  "result": true
}
```

Note the `"result": true` in the response, even though caching the module failed due to no internet connection. Deno lint afterwards still shows a `no-cache` error for the import.

It is expected that any sort of failure in fulfilling the `deno/cache` request be reflected in the response, so an error can be displayed to the user.

**Comment by kitsonk** [comment]
`deno/cache` requests are fire and forget, because they resolve asynchronously.  A failure is basically that the diagnostics don't clear.

Related feature request: #11049

**Comment by jaydenseric** [comment]
Thanks for explaining how the `deno/cache` request is currently implemented, but does it have to be that way? Is it technically impossible for the Deno language server to respond with an operation success or failure status?

The CLI `deno cache foo.js` with internet disconnected results in an error exit code 1, with an error message:

```
deno cache foo.js
Download https://deno.land/x/oak/helpers.ts
error: error sending request for url (https://deno.land/x/oak/helpers.ts): error trying to connect: dns error: failed to lookup address information: nodename nor servname provided, or not known
    at file://[redacted]/foo.js:1:0
```

**Comment by kitsonk** [comment]
> but does it have to be that way

Yes. You don't want asynchronous processes to becoming blocking ones. That is very dangerous. Also, while requesting a cache is a thing, and client should take no action if it exceeds or fails. This is the whole nature of asynchronous designs, to not assume things succeed or fail, just respond to additional messages.

The linked feature request is to send additional notifications to the client in case of a failure.

**Comment by jaydenseric** [comment]
Interestingly, I just discovered that my earlier statement that "`deno/cache requests` respond the same for success or failure" is actually untrue. Some sorts of caching errors can lead to a Deno language server internal error with a A JSON-RPC error.

For example, try caching this module (I don't actually need to use it, I just found this error while testing caching with random imports) via the Deno language server:

```
https://deno.land/x/a11yslider/src/index.ts
```

And you will see this error:

```
An unsupported media type was attempted to be imported as a module.
  Specifier: https://deno.land/x/a11yslider@v0.4.7/src/index.css
  MediaType: Unknown
Internal error
```

It looks like this to a user of my [Nova Deno extension](https://extensions.panic.com/extensions/jaydenseric/jaydenseric.deno):

<img width="337" alt="Screen Shot 2021-08-24 at 9 51 40 am" src="https://user-images.githubusercontent.com/1754873/130534221-441484eb-3ff9-452d-9174-f77a60f04a28.png">

If `deno/cache` requests are fire and forget, is that error a bug? Should all errors, not just module 404's, be silent?
